File: Backend_API/booking-clinic/src/main/java/com/example/booking_clinic/service/NotificationService.java
package com.example.booking_clinic.service;

import com.example.booking_clinic.dto.notification.NotificationResponse;
import com.example.booking_clinic.entity.User;

import java.util.List;

public interface NotificationService {
    List<NotificationResponse> getMyNotifications();

    NotificationResponse getNotificationById(Long id);

    NotificationResponse markAsRead(Long id);

    void markAllAsRead();

    void createNotification(User user, String title, String message, String type, String referenceType, Long referenceId);

    void createNotificationForAdmins(String title, String message, String type, String referenceType, Long referenceId);

    List<NotificationResponse> getAllNotifications();
}

File: Frontend_ReactJs/src/services/notificationService.js
import { apiClient } from "./apiClient";

export function getMyNotifications() {
  return apiClient("/notifications/me");
}

export function getAllNotifications() {
  return apiClient("/notifications/all");
}

export function markAsRead(id) {
  return apiClient(`/notifications/${id}/read`, { method: "PATCH" });
}

export function markAllAsRead() {
  return apiClient("/notifications/read-all", { method: "PATCH" });
}
